index.d.ts

```diff
@@ -0,0 +1,78 @@
+import { Action, ActionCreator, StoreEnhancer, Store } from 'redux';
+
+export interface StoreCreator {
+  <S, T>(
+    reducer: LoopReducer<S>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ): Store<S>;
+}
+
+export type Loop<S, A extends Action> = [S, CmdType<A>];
+
+export interface LoopReducer<S> {
```

Comment:
is there a difference between this current syntax and putting both generic params on the interface?

```js
export interface LoopReducer<S, A extends Action> {
  (state: S, action: A): S | Loop<S, A>;
}
```